src/components/ApiPublishModal/ApiPublishModal.tsx

```diff
@@ -156,14 +160,39 @@ const ApiPublishModal: React.FC = () => {
           }
         }
 
-        newApiContent = {name, namespace, openapi: parsedOpenApi};
+        newApiContent = {
+          ...apiContent,
```

Comment:
I don't think `...apiContent` is necessary as we are already overriding all the properties after that